dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/CacheConfigManager.java

```diff
@@ -24,7 +24,7 @@
 
     private static LoadingCache<String, String> installationAccessTokenCache;
 
-    private static String jsonWebToken;
+    private static volatile String jsonWebToken;
```

Comment:
On further reflection, it would be ideal if you didn't even have this field at all, and could just pass the JWT directly after generating it. But that might be tricky to do with the caching library you're using.

Reply:
I don't know why I keep thinking about this, but, will the following scenario work?

1. Two requests that will require the jsonWebToken come in simultaneously
2. Request 1 generates a jsonWebToken and stores it here
3. Before request 1 can invoke the API using the token, request 2 overwrites the jsonWebToken
4. Request 1 invokes the API using the jsonWebToken generated by request 2
5. Request 2 invokes the API using the jsonWebToken generated by request 2.

Is this harmless, or is this an issue?

Reply:
The JWT is specific to the GitHub app itself and not the install, so it should be fine if request 1 uses the JWT from request 2.

Reply:
OK, final question, sorry I don't quite the get the whole flow -- can there be more than one GitHub app id? The JWT has the GitHub appId in it; in my above scenario if request 1 and request 2 are for different app ids, then it seems like it would matter.

If there is only 1 app id, then you're right, it doesn't matter.

Reply:
There should only ever be one app id associated with a given instance of Dockstore. I'm making a writeup of the general flow of things and putting it into the Dockstore primer.